test/modern_bpf/test_suites/syscall_enter_suite/mkdir_e.cpp

```diff
@@ -0,0 +1,40 @@
+#include "../../event_class/event_class.h"
+
+#ifdef __NR_mkdir
+TEST(SyscallEnter, mkdirE)
+{
+	auto evt_test = new event_test(PPME_SYSCALL_MKDIR_2_E);
+
+	evt_test->mark_single_64bit_syscall_as_interesting(__NR_mkdir);
+
+	evt_test->enable_capture();
```

Comment:
What if these 3 lines above become an util function?
Like
```test_enable_capture_syscall(__NR_mkdir, bool enter)` -> this function would iterate over the syscall_table to find the `__NR_mkdir` enter and exit events and automatically creates the `evt_test` on the correct event.

I think it would be less error prone, because:
* we won't have to remember to update eg: `PPME_SYSCALL_MKDIR_2_E` here, when we add a new mkdir event
* we remove a bit of boilerplate
WDYT?

Reply:
I think that the new method could return an `evt_test` that, when going out of scope, would disable capture.
So, the code would become like:
```
auto evt_test = create_event_test(__NR_mkdir, enter);
// here capture is started
...
...

delete evt_test;
// here capture is stopped
```

Reply:
If you wrap it inside a `unique_ptr` you could also just do
```
{
    unique_ptr<...>evt_test = create_event_test(__NR_mkdir, enter);
    // here capture is started
    ...
}
// here capture is disabled
```

(pseudo-code of course)

Reply:
if I understood correctly, we cannot disable the capture when we destroy the `event_test` object because we still need it to perform assertions on the capture event :thinking: maybe I'm missing something here

Reply:
Note that the capture is disabled [here](https://github.com/falcosecurity/libs/pull/484/files#diff-ca741ef3bb2d1610780fcc74ae8486439bcc86a097a3db5e501574f20cf9e914R22-R24) to avoid polluting the buffer with tons of events

Reply:
Ah i see! So we need the object but we also need to stop the capture. 
Ok makes sense then; i'd leave the capture_start/stop methods to be manually called!

Reply:
We can disable the capture also at the end of the test when the `evt_test` object has terminated its lifecycle in order to use this approach, but there are 2 downsides:
* the buffer will be polluted with further events and in every test, we have to read more events when cleaning the buffers or when searching for our event, in huge environment with tons of events per second it could be quite noisy
* I don't like very much the idea of completely hiding when the capture starts and ends, but this is just a personal taste

Reply:
Ok ! right now I will add the search in the table and clear a little bit the 2 methods we have said previously